### DISCUSSION TITLE
Mock atom return value for custom hook testing

### DISCUSSION THREAD

**Discussion by ninoman** [pmndrs/jotai#1640] [Q&A] [answered]
Hi, In the testing section of the documentation there is info about how to mock atom values for the components, by wrapping them inside of the Provider and set the initial values.

But how should we do it for the hooks? I was trying the following option

`
(myAtom.read as jest.Mock).mockReturnValue('myValue');
`

but whenever you mock return value for your atom it is cached and you can't change it anymore as the atom keeps returning the cached value

`
(myAtom.read as jest.Mock).mockReturnValue('myValue');
(myAtom.read as jest.Mock).mockReturnValue('myNewValue'); // it will still return 'myValue'
(myAtom.read as jest.Mock).mockReturnValue('myOtherValue'); //it will still return 'myValue'
`

So my question is how can we reset the cache during the tests, or what is the best way to mock atom values for the custom hook testing?

**Comment by dai-shi** [comment, marked as answer]
> how can we reset the cache during the tests

We should use `Provider` and remount for each test.

**Reply by ninoman** [reply]
Yeah, I ended up doing that.

`
renderHook(useMyHook, {
  wrapper: ({ children }) => (
    <Provider initialValues={[[myAtom, true]]}>{children}</Provider>
  ),
});
`
